src/test/java/com/prgrms/board/service/PostServiceTest.java

```diff
@@ -0,0 +1,155 @@
+package com.prgrms.board.service;
+
+import com.prgrms.board.domain.Member;
+import com.prgrms.board.domain.Post;
+import com.prgrms.board.dto.PostCreateDto;
+import com.prgrms.board.dto.PostResponseDto;
+import com.prgrms.board.dto.PostUpdateDto;
+import com.prgrms.board.repository.MemberRepository;
+import com.prgrms.board.repository.PostRepository;
+import lombok.extern.slf4j.Slf4j;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.transaction.annotation.Transactional;
+
+import javax.servlet.http.HttpSession;
+
+import java.util.Optional;
+
+import static com.prgrms.board.service.PostServiceImpl.SESSION_MEMBER;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+@SpringBootTest
+@Slf4j
+class PostServiceTest {
+    @Autowired
+    PostService postService;
+
+    @Autowired
+    PostRepository postRepository;
+
+    @Autowired
+    MemberRepository memberRepository;
+
+    @Autowired
+    MemberService memberService;
+
+    private Member savedMember;
+    private Post savedPost;
+
+    @BeforeEach
+    void setup() {
+        Member member = Member.builder()
+                .name("기서")
+                .age(26)
+                .hobby("농구")
+                .build();
+
+        savedMember = memberRepository.save(member);
+
+        Post post = Post.builder()
+                .title("title")
+                .content("this is content")
+                .writer(member)
+                .build();
+
+        savedPost = postRepository.save(post);
+    }
+
+    @AfterEach
+    void clear() {
+        postRepository.deleteAll();
+        memberRepository.deleteAll();
+    }
+
+    @Test
+    @DisplayName("저장되지 않은 회원이 게시글을 저장할 경우 예외가 발생한다.")
+    @Transactional
+    void 게시글_저장_실패() {
+        //given
+        Long unknownId = 100L;
+        PostCreateDto createDto = PostCreateDto.builder()
+                .title("title1")
+                .content("content1")
+                .writerId(unknownId)
+                .build();
+
+        //when & then
+        assertThrows(RuntimeException.class, () -> postService.register(createDto));
+    }
+
+    @Test
+    @DisplayName("게시글을 단건 조회할 때 Member의 정보와 함께 조회할 수 있다.")
+    @Transactional
+    void 게시글_조회_성공() {
+        //given
+        PostResponseDto postResponseDto = postService.findById(savedPost.getId());
+        long count = postRepository.count();
+
+        assertThat(postResponseDto.getTitle()).isEqualTo(savedPost.getTitle());
+        assertThat(postResponseDto.getContent()).isEqualTo(savedPost.getContent());
+        assertThat(postResponseDto.getWriter()).isEqualTo(savedMember.getName());
```

Comment:
```suggestion
        assertThat(postResponseDto)
            .hasFieldOrPropertyWithValue("title", savedPost.getTitle())
            .hasFieldOrPropertyWithValue("content", savedPost.getContent())
            .hasFieldOrPropertyWithValue("writer", savedMember.getName());
```
이런 방법도 있답니다.
각각의 장단점은 생각해 봐야겠네요.

Reply:
@geonwoo0215 
건우님이 알려주신  방법의 가독성이 더 좋네요!
다만 두 방법 다 각각의 필드마다 검증하다보니, 새로운 필드가 생긴다면
검증 메소드를 한 줄씩 추가해야 한다는 점이 좀 걸리네요..ㅜ